packages/xrpl/src/models/ledger/Ledger.ts

```diff
@@ -63,3 +58,39 @@ export default interface Ledger {
    */
   transactions?: Array<Transaction & { metaData?: TransactionMetadata }>
 }
+
+/**
+ * A ledger is a block of transactions and shared state data. It has a unique
+ * header that describes its contents using cryptographic hashes.
+ *
+ * @category Ledger Entries
+ */
+export interface Ledger extends BaseLedger {
+  /**
+   * The ledger index of the ledger. Represented as a native JSON number.
+   */
+  ledger_index: number
```

Comment:
In hindsight, I like your other design better. Since this is the only aspect of difference between the two versions, couldn't we template based on the `APIVersion` ? This could reduce a lot of boiler-plate code downstream.

```suggestion
export interface Ledger<APIVersion> extends BaseLedger {

ledger_index : (APIVersion extends RIPPLED_API_V2 ? number : string)
}

// now all the downstream classes can just inherit from "Ledger<APIVersion>", instead of having to make two copies
// Ex
export interface LedgerBinary<APIVersion>
  extends Omit<Omit<Ledger<APIVersion>, 'transactions'>, 'accountState'> {
  accountState?: string[]
  transactions?: string[]
}
```

I'm not asking you to change the code, I was wondering if this approach has any demerits?

Reply:
I'd rather keep the design consistent of having separate interfaces for each api version: `Ledger` for v2, `LedgerV1` for v1. Thus, I think it's better to keep it as is.